src/h5web/vis-packs/nexus/utils.ts

```diff
@@ -20,35 +21,29 @@ import {
   assertStringType,
   assertArray,
   isDefined,
+  isGroup,
 } from '../../guards';
 import type { NxData, SilxStyle } from './models';
 import { isScaleType } from '../core/utils';
 import { getAttributeValue, getChildEntity } from '../../utils';
 
-export function getDatasetLabel(dataset: Dataset): string {
-  const longName = getAttributeValue(dataset, 'long_name');
-  if (longName && typeof longName === 'string') {
-    return longName;
-  }
-
-  const units = getAttributeValue(dataset, 'units');
-  if (units && typeof units === 'string') {
-    return `${dataset.name} (${units})`;
-  }
-
-  return dataset.name;
-}
-
-export function isNxDataGroup(group: Group) {
+export function isNxDataGroup(group: Group): boolean {
   return getAttributeValue(group, 'NX_class') === 'NXdata';
 }
 
-export function assertNxDataGroup(group: Group) {
+export function assertNxDataGroup(group: Group): void {
   if (!isNxDataGroup(group)) {
     throw new Error('Expected NXdata group');
   }
 }
 
+export function isNxGroup(entity: Entity): boolean {
+  return (
+    isGroup(entity) &&
+    (isNxDataGroup(entity) || !!getAttributeValue(entity, 'default'))
```

Comment:
I think it should also check that the entity has a valid `NX_class` attribute.

Someone might create HDF entities with attributes called `default` but that have nothing to do with NeXus.

Reply:
This would change the behaviour for detecting NeXus visualizations. Are you sure?

Reply:
I think it's good idea, just wanted to confirm.

Reply:
It should not change anything for properly formatted NeXus file right ? `default` attributes are always on valid NX entities.

Reply:
Yes, but if I recall, we had made the decision to be more lenient towards wrongly-formatted files (e.g. missing `NXroot` class on the root entity).

Reply:
Oh, yes, I remember now ! Then, forget it :smile: